src/live_restore/live_restore_fs.c

```diff
@@ -308,8 +314,12 @@ __live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_sessi
             }
     }
 
-    /* FIXME-WT-14040: This is a also a big old race. */
-    if (lr_fs->finished)
+    /*
+     * Once we're past the background migration stage we never need to access the source directory
+     * again.
+     */
+    WTI_LIVE_RESTORE_STATE state = __wti_live_restore_get_state(session, lr_fs);
+    if (state > WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION)
```

Comment:
I haven't seen us compare enum values with a `>` in the code base before. Let me know if you object or think this isn't clear. I prefer this way, but can swap it out for `state == WTI_LIVE_RESTORE_CLEAN_UP || state == WTI_LIVE_RESTORE_COMPLETE` if needed.

Reply:
SGTM. I've moved to the macro in https://github.com/wiredtiger/wiredtiger/pull/11563/commits/048b8f44eb03a767c7c8e6acd4b2f19d00553389. 
There's a few related comments so I'll just respond to this one, but comments with a 👍 react mean I've read and addressed them.